tests/run-make/extern-flag-fun/rmake.rs

```diff
@@ -0,0 +1,36 @@
+// The --extern flag can override the default crate search of
+// the compiler and directly fetch a given path. There are a few rules
+// to follow: for example, there can't be more than one rlib, the crates must
+// be valid ("no-exist" in this test), and private crates can't be loaded
+// as non-private. This test checks that these rules are enforced.
+// See https://github.com/rust-lang/rust/pull/15319
```

Comment:
Suggestion: we could annotate with comments for which section of the test checks which rule to help the test reader.